aiida_vasp/workflows/relax.py

```diff
@@ -0,0 +1,298 @@
+"""Structure relaxation workchain for VASP."""
+import numpy
+from aiida.common.extendeddicts import AttributeDict
+from aiida.work import WorkChain
+from aiida.work.workchain import append_, while_, ToContext
+
+from aiida_vasp.utils.aiida_utils import get_data_class, get_data_node
+from aiida_vasp.utils.vasp.isif import IsifStressFlags, Isif
+from aiida_vasp.utils.vasp.ibrion import IbrionFlags, Ibrion
+from aiida_vasp.utils.vasp.encut import EncutFlags, Encut
+from aiida_vasp.utils.vasp.nsw import Nsw
+from aiida_vasp.workflows.base import VaspBaseWf
+from aiida_vasp.workflows.restart import prepare_process_inputs, UnexpectedCalculationFailure
+
+RELAXATION_INCAR_TEMPLATE = AttributeDict({
+    "ismear": -1,
+    "sigma": 0.2,
+    "algo": 'NORMAL',
+    "ediff": 1E-6,
+    "prec": 'Normal',
+    "ibrion": 2,
+    "nsw": 200,
+    "nelmin": 4,
+    "ediffg": -0.01,
+    "isif": 4
+})
+
+
+def relax_parameters(pos, shape, volume):
+    """Set IBRION and ISIF according to the choice of what should be relaxed."""
+    if not (pos or volume or shape):
+        raise ValueError('relaxation workflow requires at least one of "positions", "cell shape", "cell volume" to be alowed to change.')
+    if not pos:
+        ibrion = Ibrion(ion_updates=IbrionFlags.NO_UPDATE)
+    else:
+        ibrion = Ibrion(ion_updates=IbrionFlags.IONIC_RELAXATION_CG)
+    isif = Isif(calculate_stress=IsifStressFlags.FULL, vary_positions=pos, vary_cell_shape=shape, vary_cell_volume=volume)
+    nsw = Nsw(value=200, ibrion=ibrion)
```

Comment:
Maybe it would make sense to make a structure where the tags was composed into a `WorkingIncar` or similar, which turns into an `Incar` when everything is setup and ready to go. As such the tag classes can access other parameters without passing them around. This would also house interdependence checking etc. But okey, let us leave this for now.

Reply:
I was thinking the same thing. The reason I went with separate classes is that I was testing how it feels to use them and that I wanted to keep tag checking "opt-in" for now, as in you have to specifically check.

But it seems to me now, that it is really sufficient to give the options to a) check everything we know to check or b) you are on your own. What do you think about this?

option a) could also have "opt-outs" for specific checks

Reply:
I understand. Yes, I agree on this.